rust/README.md

```diff
@@ -2,40 +2,52 @@
 
 Earthly's official collection of rust [UDCs](https://docs.earthly.dev/docs/guides/udc).
 
-## +CARGO
-
-This UDC runs the cargo command `cargo $args` caching the contents of `$CARGO_HOME/registry`, `$CARGO_HOME/git` and `target` for future builds of the same calling target. 
-
-### Usage
-
 First, import the UDC up in your Earthfile:
 ```earthfile
 VERSION 0.7
 IMPORT github.com/earthly/lib/rust:<version/commit> AS rust
 ```
 
-Then, just use it in your own targets and UDCs:
-```earthfile
-DO rust+CARGO ...
-```
+## +INIT
 
-### Thread safety
-This UDC should be thread safe. Parallel builds of targets using it should be free of race conditions.
+This UDC stores the configuration required by the other UDCs in the build environment filesystem, and installs required dependencies.
 
-### Arguments
+It is meant to be called once per build environment, to avoid passing repetitive arguments to the UDCs called after it, and to install required dependencies before the source files are copied from the build context.  
 
-#### `args`
-Cargo subcommand and its arguments. Required.
+### Usage
 
-#### `keep_fingerprints (false)`
-Do not remove source packages fingerprints. Use only when source packages have been `COPY`ed with `--keep-ts` option.
+Call once per build environment:
+```earthfile
+DO rust+INIT ...
+```
 
-Cargo caches compilations of packages in `target` folder based on their last modification timestamps. 
+### Arguments
+#### `cache_id`
+Overrides default ID of the global `$CARGO_HOME` cache. Its value is exported to the build environment under the entry: `$CARGO_HOME_CACHE_ID`.
 
+#### `keep_fingerprints (false)`
+Instructs the following `+CARGO` calls to don't remove the Cargo fingerprints of the source packages. Use only when source packages have been COPYed with `--keep-ts `option.
+Cargo caches compilations of packages in `target` folder based on their last modification timestamps.
 By default, this UDC removes the fingerprints of the packages found in the source code, to force their recompilation and work even when the Earthly `COPY` commands used overwrote the timestamps.
```

Comment:
TBH I'm not sure how well this is working, I think there should be a strong recommendation to use `--keep-ts` and to enable `keep_fingerprints` (clarifying the double negative: I mean that the fingerprints should be left alone).

Reply:
Agree. We will be improving the way timestamps are handled soon, and this hack will be removed.